### ISSUE TITLE
CA1823 flags [InlineArray]'s field definitions as unused

### ISSUE THREAD

**Issue by MihaZupan** [dotnet/roslyn-analyzers#6788]
### Analyzer

**Diagnostic ID**: [CA1823](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1823): `Avoid unused private fields`

### Analyzer source

The version used by dotnet/runtime.

**Version**: 8.0.0-preview.23364.2

### Describe the bug

When defining a struct with the new `[InlineArray]` attribute, the field definition may be flagged by CA1823 as unused even though it is used by the implicitly generated members.

### Steps To Reproduce

```c#
using System.Runtime.CompilerServices;

[InlineArray(8)]
internal readonly struct MyStruct
{
    private readonly object? _ref0; // CA1823 fires

    public MyStruct(ReadOnlySpan<object> values) => values.CopyTo(this!);
}

// Some consuming code elsewhere
EightPackedReferences refs = new(...);
Console.WriteLine(refs[0]);
```

### Expected behavior

`CA1823` doesn't trigger for fields on structs annotated by `[InlineArray]`.

### Actual behavior

The field is flagged as unused.

**Comment by MihaZupan** [comment]
I guess this will be covered by #6789
